presto-product-tests/src/main/resources/sql-tests/testcases/intersect/intersect.sql

```diff
@@ -0,0 +1,29 @@
+-- database: presto; requires: com.facebook.presto.tests.ImmutableTpchTablesRequirements; tables: nation, workers; groups: intersect;
```

Comment:
What about test cases that mix INTERSECT and UNION ALL?
What happens if the data types don't exactly match. Is there an implicit coercion in some cases? I'm sure there are tests for UNION [ALL].